src/windows/audio.c

```diff
@@ -206,18 +207,28 @@ static HRESULT audio_device_create(MTY_Audio *ctx)
 
 	WAVEFORMATEXTENSIBLE pwfx = {
 		.Format.wFormatTag = WAVE_FORMAT_PCM,
-		.Format.nChannels = ctx->channels,
-		.Format.nSamplesPerSec = ctx->sample_rate,
-		.Format.wBitsPerSample = AUDIO_SAMPLE_SIZE * 8,
-		.Format.nBlockAlign = ctx->channels * AUDIO_SAMPLE_SIZE,
-		.Format.nAvgBytesPerSec = ctx->sample_rate * ctx->channels * AUDIO_SAMPLE_SIZE,
+		.Format.nChannels = (WORD) ctx->cmn.format.channels,
+		.Format.nSamplesPerSec = ctx->cmn.format.sampleRate,
+		.Format.wBitsPerSample = (WORD) ctx->cmn.computed.sample_size * 8,
+		.Format.nBlockAlign = (WORD) ctx->cmn.computed.frame_size,
+		.Format.nAvgBytesPerSec = (DWORD) ctx->cmn.computed.buffer_size,
 	};
 
-	// We must query extended data for greater than two channels
-	if (ctx->channels > 2) {
-		e = audio_get_extended_format(device, &pwfx);
-		if (e != S_OK)
-			goto except;
+	if (ctx->cmn.format.channels > 2) {
+		if (ctx->cmn.format.channelMask == 0) {
+			e = audio_get_extended_format(device, &pwfx);
+			if (e != S_OK)
+				goto except;
+
+		} else {
+			pwfx.Format.wFormatTag = WAVE_FORMAT_EXTENSIBLE;
+			pwfx.Format.cbSize = 22;
```

Comment:
I know the documentation states "should be bigger than 22", I think this should be set to:
```suggestion
			pwfx.Format.cbSize = sizeof(WAVEFORMATEXTENSIBLE);
```

Reply:
Actually, 22 is correct but it is misleading. The more accurate way to write it is:
```c
pwfx.Format.cbSize = sizeof(WAVEFORMATEXTENSIBLE) - sizeof(WAVEFORMATEX);
```

Why? Because docs say that the value is the total bytes of data _appended_ to the `WAVEFORMATEX` struct. _Not the total bytes of the whole struct_.
![image](https://github.com/user-attachments/assets/91c08610-785d-461b-892e-775eab2ff2d3)
![image](https://github.com/user-attachments/assets/4376d73c-ee8b-430f-8203-7183d1fa0cab)

I have made this change now, thanks.